src/lib/db.js

```diff
@@ -77,17 +77,29 @@ export class Database {
     return this.db.creator.get({ url: url });
   }
 
-  async getCreators({ limit = 100, withDurations = false } = {}) {
+  async getCreators({
+    limit = 100,
+    withDurations = false,
+    withThanksAmount = false,
+  } = {}) {
     let creators = await this.db.creator.limit(limit).toArray();
     if (withDurations) {
-      await Promise.all(
+      creators = await Promise.all(
```

Comment:
You're actually modifying the creators in place here so assigning them looks like a lie to me. If you were to copy them this would be fine but you're not. Either just await the Promise.all without assigning to creators (the mental model is, you map from creators to promises, you're not actually modifying creators, you give up the promises to the runtime and the runtime modifies the creators, [ugly](https://en.wikipedia.org/wiki/JavaScript)) or you do what you're really implying when you use `map`, i.e. you copy.

Reply:
I think I discussed this with @ErikBjare and we thought this was the best option but I don't know if I agree now. I can think of a few options but I'll just revert to be able to merge.